### PR TITLE
AuthAddressLogic: AWA shutdown and recovery

### PR DESCRIPTION
**Pull request by goatpig**
fix AuthAddressLogic shutdown sequence

### CODE REVIEW

File: BlocksettleNetworkingLib/AuthAddressLogic.cpp [hunk 1]
```diff
@@ -289,14 +302,17 @@ void AuthAddressValidator::waitOnRefresh(const std::string& id)
...
+         auto&& notifId = refreshQueue_.pop_front(std::chrono::seconds{ 5 });
+         if ((notifId == idRef) || stopped_) {
             break;
          }
       }
       catch (const ArmoryThreading::StopBlockingLoop&) {
          break;
       }
+      catch (const ArmoryThreading::StackTimedOutException &) {
+         break;
```
Comment (goatpig) [inline BlocksettleNetworkingLib/AuthAddressLogic.cpp:314]: This will kill the wait in 5 if pop_front times out, it should be continue, not break

Comment (sergey-chernikov) [reply BlocksettleNetworkingLib/AuthAddressLogic.cpp]: True, made it conditional

Comment (goatpig) [reply BlocksettleNetworkingLib/AuthAddressLogic.cpp]: The common side changes work, need to conclude the conversation on the PB test side.

Comment (goatpig) [reply BlocksettleNetworkingLib/AuthAddressLogic.cpp]: This is ready to go

**Pull request merged**